File: cpi.py
#!/usr/bin/python
#from decimal import *
#precision = 11
#getcontext().prec = precision
n = 1000000
def f(x):
    return (1-(float(x)**2))**float(0.5)
val = 0
for i in range(n):
    i = i+1
    val = val+f(float(i)/float(n))
val = val*2
pi = (float(2)/n)*(float(1)+val)
print pi

File: cpi.pl
#!/usr/bin/perl
$n = '1000000';
$n = $n + 0;
$h = $n * 0;
for($a = 1; $a < $n; $a = ++$a){
    $t = $a/$n;
    $val = (1-(($t)**2))**0.5;
    $h+=$val;
}
$h = $h*2;
$pi = (2/$n)*(1+$h);
printf "%.11f", $pi;
